infra/src/test/kotlin/fi/fta/geoviite/infra/ui/pagemodel/map/PreviewChangesElement.kt

```diff
@@ -40,6 +41,31 @@ class E2EChangePreviewTable(
         return E2EMenu()
     }
 
+    fun closeMenu(change: E2EChangePreviewRow) {
+        logger.info("Close more menu for change $change")
+
+        selectBy(change, byQaId("menu-button"))
+    }
+
+    fun movePublicationGroup(publicationGroup: PublicationGroup): E2EChangePreviewTable = apply {
+        rows.find { row ->
+            val rowMenu = openMenu(row)
+
+            val movePublicationGroupMenuItem = rowMenu.items.firstOrNull { menuItem ->
+                val menuItemQaId = menuItem.element?.getAttribute("qa-id")
+                menuItemQaId == "preview-move-publication-group-${publicationGroup.id}"
+            }
+
+            movePublicationGroupMenuItem?.let { item ->
+                item.element?.click()
+                true
+            } ?: run {
+                closeMenu(row)
+                false
+            }
+        }
+    }
```

Comment:
Sinällään vähän hähmeä rakenne toteuttaa tuo "etsi rivi joka toteuttaa ominaisuuden", "tee sille operaatio x" ja "lopeta haku".

Normaalilla for-loopilla vaan breakkaisi/returnaisi ulos loopista, mutta meillä eivät nämä oikein ole olleet käytössä.

Reply:
Voi meillä `while`-, `for`- jne. rakenteita käyttää jos näkee niille järkeviä käyttökohteita (ts. ei olla kategorisesti bannattukaan niitä.) Yleensä niitä vaan tuppaa näkemään lähinnä tilanteissa joissa halutaan käydä aina koko lista läpi, ja sen saa toteutettua helpommin jollain higher order -kamalla. Tässä voisi olla OK kohta ihan käsin ohjatullekin perusloopille (ja tuo `.find()`-rakenne on tosiaan melko jännä, joten vähintään joku tyyliin `.foreach()` + `return` voisi olla selkeämpi rakenne jos et haluakaan mennä ihan raakaan looppiin)